docs/decisions/20240116-use-zio-failures-and-defects-effectively.md

```diff
@@ -199,24 +197,16 @@ As a user of an existing component or service, you should exclusively catch fail
 effectively handle. Any unhandled failures should be transformed into defects and propagated through the call stack. You
 should not expect callers of your component to handle lower-level failures that you do not handle.
 
-The conversion of a ZIO failure into a ZIO defect can be achieved using **ZIO#orDie** and **ZIO#refineOrDie**.
-
 #### Use Failure Wrappers To Prevent Failures Leakage From Lower Layers
 
-When invoking lower-level components, refrain from directly exposing their failure types in your component or service
-interface.
-
-Using failures wrappers serves to encapsulate and abstract failure types originating from lower-level components or
-services. It prevents the direct exposure of these lower-level failure types in the interface of higher-level components
-or services, thus enhancing loose coupling and safeguarding against leakage of underlying implementation details to the
-caller.
+When invoking lower-level components, do not directly expose their failure types in your component interface. Use
+wrappers to encapsulate and abstract failure types originating from lower-level components, thus enhancing
+loose coupling and safeguarding against leakage of underlying implementation details to the caller.
 
-While using failure wrappers to shield callers from lower-level failures is beneficial for abstraction and loose
-coupling, **it should not be the default failure-handling strategy**. Lower-level failures should primarily be managed
-at the component implementation level ensuring that the component handles and appropriately resolves and recovers them.
+Using failure wrappers and propagate them **should not be the default strategy**. Lower-level failures should primarily
+be managed at your component implementation level ensuring that it handles and appropriately recovers them.
 
-Unhandled failures within the component's boundaries should preferably be transformed into defects rather than
-propagated upward.
+Unhandled failures within the component's boundaries should preferably be transformed into defects.
```

Comment:
```suggestion
Handling failures within the component's boundaries is preferred by transforming them into defects whenever possible.
```

Reply:
I think your suggestion does not convey the initial message. Would something like this be better?

_"Failures not handled within the component's boundaries should preferably be transformed into defects whenever possible."_

(note: on the other hand, those that are handled should not be transformed to defects)

Reply:
That would work - thanks for double checking